### ISSUE TITLE
Enhancement: [@typescript-eslint/strict-boolean-expressions] allowNullableEnum option

### ISSUE THREAD

**Issue by Hamusutaa** [typescript-eslint/typescript-eslint#5614]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/strict-boolean-expressions/

### Description

I propose that a new option be added to the `@typescript-eslint/strict-boolean-expressions` rule: `allowNullableEnum`, which would behave similarly to the `allowNullableNumber` option, but for Enums which contain a zero as one of their possible values.  It could default to true in order to remain backwards-compatible, but allow being set false in order to catch potential issues where one of the valid values of the Enum is falsy.

### Fail

```typescript
enum ExampleEnum {
  This = 0,
  That = 1,
}

const rand = Math.random();

let theEnum: ExampleEnum | null = null;
if (rand < .3) {
  theEnum = ExampleEnum.This;
} else if (rand < .7) {
  theEnum = ExampleEnum.That;
}

// No warning thrown
if (theEnum) {
  console.log('That!');
}
```

### Pass

```typescript
enum ExampleEnum {
  This = 0,
  That = 1,
}

const rand = Math.random();

let theEnum: ExampleEnum | null = null;
if (rand < .3) {
  theEnum = ExampleEnum.This;
} else if (rand < .7) {
  theEnum = ExampleEnum.That;
}

if (theEnum === ExampleEnum.That) {
  console.log('That!');
}
```

**Comment by kazizi55** [comment]
I would like to work on this issue!

**Comment by JoshuaKGoldberg** [comment]
![Captain Picard saying "make it so"](https://user-images.githubusercontent.com/3335181/192101276-4b0889a7-c3c3-4df8-8ce7-02c40e36971d.gif)
